internal/controller/git/git.go

```diff
@@ -404,24 +447,121 @@ func (r *repo) LastCommitID() (string, error) {
 	return strings.TrimSpace(string(shaBytes)), nil
 }
 
-func (r *repo) ListTags() ([]string, error) {
-	if _, err :=
-		libExec.Exec(r.buildGitCommand("fetch", "origin", "--tags")); err != nil {
+func (r *repo) ListTagsWithMetadata() ([]TagMetadata, error) {
+	if _, err := libExec.Exec(r.buildGitCommand("fetch", "origin", "--tags")); err != nil {
 		return nil, fmt.Errorf("error fetching tags from repo %q: %w", r.url, err)
 	}
-	tagsBytes, err := libExec.Exec(r.buildGitCommand("tag", "--list", "--sort", "-creatordate"))
+
+	// These formats are quite complex, so we break them down into smaller
+	// pieces for readability.
+	//
+	// They are designed to output the following fields, separated by `|*|`:
```

Comment:
I am curious about the choice of `|*|` as a delimiter. I'm not suggesting it should change. Asking for my own benefit.

Reply:
I wanted something that was _highly unlikely_ to show up in some of the fields, I could think of reasons e.g. `,` or `|` would show up. But I would be quite baffled about this happening for `|*|`.

Reply:
(The reason it needs to be a (set of) character(s) is that unlike `git log` formatting, it `git for-each-ref` can't insert tabs or other whitespace-like escape sequences.)